include/odp/api/spec/reassembly.h

```diff
@@ -0,0 +1,120 @@
+	odp_bool_t ip;
+
+	/** Reassembly offload for IPv4 packets only */
+	odp_bool_t ipv4;
+
+	/** Reassembly offload for IPv6 packets only */
+	odp_bool_t ipv6;
+
+	/** Maximum time in ns that a fragment can wait in the reassembly
+	 *  offload for the arrival of further fragments.
+	 */
+	uint64_t max_wait_time;
+
+	/** Maximum number of fragments that can be reassembled */
+	uint16_t max_num_frags;
+
+} odp_reass_capability_t;
+
+/**
+ * Fragment reassembly configuration
+ *
+ * Configure inline fragment reassembly offload support. Fragment
+ * reassembly offload can be enabled in IPSEC and PKTIN operations.
+ *
+ * When the offload is enabled, fragments will be delayed for a specified time
+ * period to allow reassembly.
+ *
+ * Reassembly result will be delivered to the application through an ODP packet
+ * with reassembly metadata. odp_packet_reass_status() can be used to query if
+ * reassembly has been attempted and if reassembly was successfully comnpleted.
+ *
+ * In case of successful reassembly, the reassembled packet is delivered
+ * to the receiver as a regular ODP packet as if the reassembled packet
+ * was received as such. When reassembly is enabled in pktio, it will be
+ * be attempted before other offloads such as packet parsing, inline IPsec and
+ * classification.
+ *
+ * In case of failed reassembly, the result is delivered to the application
+ * as a special packet that does not contain valid packet data. Such a
+ * packet can be used to get information of the incomplete reassembly
+ * so that the application can try to retry or continue the reassembly.
+ * See odp_packet_reass_partial_state().
+ *
+ * Reassembly may not complete if not all fragments were received in time but
+ * also for packet parsing difficulty, fragment overlap, resource shortage or
+ * other reasons. In such cases, application may receive packets with reassembly
+ * status as either ``ODP_PACKET_REASS_NONE`` or ``ODP_PACET_REASS_INCOMPLETE``.
+ *
+ * This structure is used only for configuration, not for capability
+ * query even though it is indirectly contained in odp_pktio_capability_t.
+ * The content of odp_pktio_capability_t.config.reassembly written by
+ * odp_pktio_capability() is undefined. Reassembly capabilities of a pktio
+ * can be checked through odp_pktio_capability_t.reassembly.
+ *
+ * @see odp_packet_reass_status(), odp_packet_reass_partial_state()
+ */
+typedef struct odp_reass_config_t {
+	/** Attempt inline reassembly of IPv4 packets. Disabled by default */
+	odp_bool_t en_ipv4;
+
+	/** Attempt inline reassembly of IPv6 packets. Disabled by default */
+	odp_bool_t en_ipv6;
+
+	/** Maximum time in ns that a fragment may wait in the reassembly
+	 *  offload for the arrival of further fragments. The value may
+	 *  not exceed the max_wait_time capability.
+	 *
+	 *  Default value is 0.
+	 */
+	uint64_t max_wait_time;
+
+	/** Maximum number of fragments that can be reassembled
+	 *
+	 * Default value is 0.
```

Comment:
Wouldn't it be better to specify: "Minimum allowed value is 2. Default value is 2."

Reply:
I agree on the minimum value that a reassembly capable implementation would support. But I wasn't sure what would be the default value for a non-reassembly capable implementation. So just left it as 0 since implementation can do a memset.

Do you recommend making the default value as 2?

Reply:
I was thinking that defining the minimum as 2 (and therefore default cannot be zero either) avoids the problem of having to think what values 0 and 1 exactly mean. And it leaves 0 free for possible future use.

The default value of this field should not matter for non-reassembly-capable implementations since in that case en_ipv4 and en_ipv6 have to be false.

Reply:
Understood. Updated so in v8.